config/routes.rb

```diff
@@ -124,10 +124,11 @@
       post 'attach_captions'
       get :captions
       get :waveform
-      get 'adaptive', to: 'master_files#hls_adaptive_manifest'
+      match ':quality.m3u8', to: 'master_files#hls_manifest', via: [:get], as: :hls_manifest
       get 'structure', to: 'master_files#structure', constraints: { format: 'json' }
       post 'structure', to: 'master_files#set_structure', constraints: { format: 'json' }
       delete 'structure', to: 'master_files#delete_structure', constraints: { format: 'json' }
+      get 'auth_token', to: 'master_files#auth_token'
```

Comment:
When you do move this route, can we consider a more descriptive name like `iiif_auth_token` or `iiif_token`?